modules/light-clients/06-solomachine/misbehaviour.go

```diff
@@ -71,6 +71,24 @@ func (sd SignatureAndData) ValidateBasic() error {
 	return nil
 }
 
+// ValidateBasic ensures that the signature and data fields are non-empty.
+func (sd SignatureAndDataV2) ValidateBasic() error {
+	if len(sd.Signature) == 0 {
```

Comment:
This is a nit, but maybe UX could be slightly improved if we group errors, in case more than one thing is empty. For example, if both signature and data are empty, then you can return one error saying that `signature and data cannot be empty`.

Reply:
i kind of like having the separate signatures rather than arbitrarily grouping errors (sometimes two? sometimes 3?), this way we can keep it consistent across the codebase...

Reply:
I think this point has come up a few times before. I tend to agree with the general behaviour of the sdk of "fail fast" where the error is returned immediately rather than grouping. I think it makes for more idiomatic go code. I do see the reasoning for validating the entire message and returning a group of error responses... it is in line with more traditional rest api design in a way. But all things considered I'd prefer to keep the code consistent and stay aligned with the sdk pattern.

If we were to change the behaviour here it should be changed across our entire codebase. And still, it would differ from standard sdk modules then like `x/auth`,`x/bank`, `x/gov` ..etc